src/Header/Header.jsx

```diff
@@ -77,12 +77,48 @@ export class Header extends Component {
 		}
 	}
 
+	dropdownToggleHandler(idOfOpenDropdown) {
+		// If we've got an onDropdownOpen or onDropdownClose prop call the onDropdownOpen or onDropdownClose callback based on idOfOpenDropdown.
+		const { onDropdownOpen, onDropdownClose } = this.props;
+
+		if (onDropdownOpen) {
+			const onDropdownOpenCallback =
+				typeof onDropdownOpen === "function"
+					? onDropdownOpen
+					: window[onDropdownOpen];
+			if (
+				idOfOpenDropdown !== null &&
+				typeof onDropdownOpenCallback === "function"
+			) {
+				onDropdownOpenCallback();
+				return true;
+			}
+		}
+
+		if (onDropdownClose) {
+			const onDropdownCloseCallback =
+				typeof onDropdownClose === "function"
+					? onDropdownClose
+					: window[onDropdownClose];
+			if (
+				idOfOpenDropdown == null &&
```

Comment:
Makes sense for the id check to be in the outer if I think. Same for the open state above